### PR TITLE
Arduino signed updater

### PR DESCRIPTION
**Pull request by dirkx**
Not so much as a 'real pull request' -- but more as I am not sure how to best start the conversation as to if this is useful for more people.

What this contains is the ability to use public-key encryption (standard X.509 / RFC 3161 digital timestamps/signatures) to secure an over the air update.

See https://github.com/dirkx/arduino-esp32/blob/arduino-signed-updater/tools/digital-signing.md for an explanation of what is going on.

It also refactors the ArduinoOTA and updater code to allow more modern Hashes.

### CODE REVIEW

**Comment by me-no-dev** [comment]
the amount of code is really overwhelming. I quite like the end result though. looking forward giving this a try for 2.0

**Comment by dirkx** [comment]
Yes - I apologies for that. Hope I can move the bulk ultimately into MBED TLS. That helps.

**Comment by me-no-dev** [comment]
I understand that it requires quite a bit of code :) also you changed the server architecture a bit, which was also nice. I did not see, is the readme included in the changes?

**Comment by me-no-dev** [comment]
Moving the mbedtls code would mean ESP-IDF or the lib-builder that compiles the libs from esp-idf.

**Comment by me-no-dev** [comment]
in `espota.py` I notice shell commands being executed (openssl, curl). Also some hardcoded values. Should I presume the changes to `espota.py` to be WIP?

**Comment by dirkx** [comment]
Well - pretty much all of it is a WIP - as said in the intro  - this was more for discussion than for immediate inclusion.

But happy to make it tree ready in the next few days.

**Comment by me-no-dev** [comment]
ok, let's start one by one :)

Maybe from the functions that espota will need and how to make them pure python (so it can be then compiled into EXE for Windows and binary for Mac)

File: libraries/HTTPUpdate/src/HTTPUpdate.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /**
  *
- * @file HTTPUpdate.cpp based om ESP8266HTTPUpdate.cpp
+ * @file HTTP_updater->cpp based om ESP8266HTTP_updater->cpp
```
Comment (Jorgen-VikingGod) [inline libraries/HTTPUpdate/src/HTTPUpdate.cpp:3]: search replace also renamed this comment...

**Comment by torntrousers** [comment]
> Is this still relevant? If so, what is blocking it? Is there anything you can do to help move it forward?
> 
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

Would be nice to have this feature though

**Comment by lucasssvaz** [comment]
@dirkx What is missing in this PR ? Could you also fix the merge conflicts ?

**Comment by me-no-dev** [comment]
let's just close it. it's draft and @dirkx can open a new one on a fresh base

**Pull request closed without merging**